codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/protocol/aws/AwsJson10ProtocolGenerator.java

```diff
@@ -49,14 +54,22 @@ public void generateRequestSerializers(GenerationContext context) {
         var writer = context.getWriter().get();
         var model = context.getModel();
         var ops = model.getOperationShapes();
+        Set<Shape> shared = new java.util.HashSet<>(Set.of());
         for (var op : ops) {
-            requestSerializerCode(op, writer);
```

Comment:
You can use two loops here - keep these logically separate, the cost of double iteration is insignificant in this context.